firmware/fsm.c

```diff
@@ -628,6 +653,47 @@ void fsm_msgGetAddress(GetAddress *msg)
 	layoutHome();
 }
 
+void fsm_msgEthereumGetAddress(EthereumGetAddress *msg)
+{
+	RESP_INIT(EthereumAddress);
+
+	if (!storage_isInitialized()) {
+		fsm_sendFailure(FailureType_Failure_NotInitialized, "Device not initialized");
+		return;
+	}
+
+	if (!protectPin(true)) {
+		layoutHome();
+		return;
+	}
+
+	const HDNode *node = fsm_getDerivedNode(SECP256K1_NAME, msg->address_n, msg->address_n_count);
+	if (!node) return;
+
+	resp->address.size = 20;
+
+	if (!hdnode_get_ethereum_pubkeyhash(node, resp->address.bytes))
+		return;
+
+	if (msg->has_show_display && msg->show_display) {
+		char desc[16];
+		strlcpy(desc, "Address:", sizeof(desc));
+
+		char address[41];
+		data2hex(resp->address.bytes, 20, address);
+
+		layoutAddress(address, desc);
```

Comment:
Yes, changed in https://github.com/trezor/trezor-mcu/pull/103/commits/48008ddd8eeae19005011dc01a3d9bc24c1be244